search/lib/api/wfs.js

```diff
@@ -6,75 +6,95 @@ const { assertValid, schemas } = require('../validator');
+    const currPage = parseInt(extractParam(event.queryStringParameters, 'page_num', '1'), 10);
+    const nextPage = currPage + 1;
+    const prevPage = currPage - 1;
+    const newParams = { ...event.queryStringParameters } || {};
+    newParams.page_num = nextPage;
+    const newPrevParams = { ...event.queryStringParameters } || {};
+    newPrevParams.page_num = prevPage;
+    const prevResultsLink = generateAppUrlWithoutRelativeRoot(event, event.path, newPrevParams);
+    const nextResultsLink = generateAppUrlWithoutRelativeRoot(event, event.path, newParams);
 
-  const provider = request.params.providerId;
-  const params = Object.assign(
-    { provider_short_name: provider },
-    cmr.convertParams(cmr.WFS_PARAMS_CONVERSION_MAP, request.query)
-  );
-  const collections = await cmr.findCollections(params);
-  const collectionsResponse = {
-    id: provider,
-    stac_version: settings.stac.version,
-    description: `All collections provided by ${provider}`,
-    license: 'not-provided',
-    links: [
-      wfs.createLink('self', generateAppUrl(event, `/${provider}/collections`),
-        `All collections provided by ${provider}`),
-      wfs.createLink('root', generateAppUrl(event, '/'), 'CMR-STAC Root'),
-      {
-        rel: 'next',
-        href: nextResultsLink
-      }
-    ],
-    collections: collections.map(coll => convert.cmrCollToWFSColl(event, coll))
-  };
+    const provider = request.params.providerId;
+    const params = Object.assign(
+      { provider_short_name: provider },
+      cmr.convertParams(cmr.WFS_PARAMS_CONVERSION_MAP, request.query)
+    );
+    const collections = await cmr.findCollections(params);
+    if (!collections.length) throw new Error('Collections not found');
+    const collectionsResponse = {
+      id: provider,
+      stac_version: settings.stac.version,
+      description: `All collections provided by ${provider}`,
+      license: 'not-provided',
+      links: [
+        wfs.createLink('self', generateAppUrl(event, `/${provider}/collections`),
+          `All collections provided by ${provider}`),
+        wfs.createLink('root', generateAppUrl(event, '/'), 'CMR-STAC Root'),
+        {
+          rel: 'next',
+          href: nextResultsLink
+        }
+      ],
+      collections: collections.map(coll => convert.cmrCollToWFSColl(event, coll))
+    };
 
-  if (currPage > 1 && collectionsResponse.links.length > 1) {
-    collectionsResponse.links.splice(collectionsResponse.links.length - 1, 0, {
-      rel: 'prev',
-      href: prevResultsLink
-    });
-  }
+    if (currPage > 1 && collectionsResponse.links.length > 1) {
+      collectionsResponse.links.splice(collectionsResponse.links.length - 1, 0, {
+        rel: 'prev',
+        href: prevResultsLink
+      });
+    }
+
+    if (collectionsResponse.collections.length < 10) {
+      collectionsResponse.links.splice(collectionsResponse.links.length - 1);
+    }
```

Comment:
- Yeah, that's a good point- the if the order of the links changes then this could cause problems. I think I'll go with what you suggested and change it from removing the link to conditionally adding it.